messages/eth.proto

```diff
@@ -56,6 +56,7 @@ message ETHSignRequest {
   AntiKleptoHostNonceCommitment host_nonce_commitment = 9;
   // If non-zero, `coin` is ignored and `chain_id` is used to identify the network.
   uint64 chain_id = 10;
+  string address = 11;
```

Comment:
i am now thinking that it might be better to provide a `enum ETHAddressCase { MIXED = 0; LOWER = 1; UPPER = 2; }` instead and format the address to be displayed/return according to this. 

The benefit is that it is explicit. An address string can contain many invalid values while the enum can only have valid values.

What do you think? the downside i could see is that figuring out which value to provide would then be offloaded to the client libraries (or their users), but it's generally better to have libraries deal with such complexities than the firmware.

Reply:
That's an okay idea. But figuring out which value to provice you mention confuses me. Then how are we supposed to interpret user input in the code? Mixed, lower or upper? Do we check all 3 cases and then construct `ETHAddressCase` from whichever that suits? Do we just throw an `InvalidInputError` when it's not a valid string?

Reply:
the enum would be instead of the address (the new address field would be dropped again). the client library then simply takes the enum value as a method argument in `eth_sign()`.